src/AzureAD.tsx

```diff
@@ -58,61 +54,53 @@ class AzureAD extends React.Component<IProps, IState> {
   constructor(props: IProps) {
     super(props);
 
-    const authenticationState = AuthenticationState.Unauthenticated;
-
     this.authProvider = this.props.provider.getAuthProvider();
-    this.authProvider.userInfoChangedCallback = this.updateState;
+    this.authProvider.onAuthenticationStateChanged = this.updateAuthenticationState;
 
-    this.state = { authenticationState };
+    this.state = { authenticationState: this.authProvider.authenticationState }
   }
 
   public render() {
     switch (this.state.authenticationState) {
       case AuthenticationState.Authenticated:
-        return this.props.authenticatedFunction(this.logout);
+        if (this.props.authenticatedFunction) {
+          return this.props.authenticatedFunction(this.logout);
+        }
+        else {
+          return this.props.children;
```

Comment:
This fills a scenario my team found useful. If no authenticated callback is given, just display whatever elements is inside the `AzureAD` component as children.